pandas/io/stata.py

```diff
@@ -1327,12 +1327,20 @@ def _calcsize(self, fmt):
 
     def _decode(self, s):
         s = s.partition(b"\0")[0]
-        return s.decode('utf-8')
+        try:
+            return s.decode('utf-8')
+        except UnicodeDecodeError:
+            # GH 25960
+            return s.decode('latin-1')
```

Comment:
Should we wrap this line in a `try-except` and provide a nicer error message regarding failure to decode?  Same question regarding the `try-except` you have for encoding.

Reply:
It is rarely hit -- and only on an incorrectly formatted file, so I would say no.  There is nothing the user can realistically do to overcome a decode error now.

Reply:
Sure, but is the error message that you would get user-friendly?  The goal of the `try-except` that I am asking about is to produce a nicer error message, not necessarily to "overcome" the error.

Reply:
I don't think this ever fails since 'latin-1' can decode all bytes.

Reply:
```
In [24]: b''.join([chr(i).encode('latin-1') for i in range(256)]).decode('latin1')

Out[24]: '\x00\x01\x02\x03\x04\x05\x06\x07\x08\t\n\x0b\x0c\r\x0e\x0f\x10\x11\x12\x13\x14\x15
\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f !"#$%&\'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQR
STUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~\x7f\x80\x81\x82\x83\x84\x85\x86\x87\x88\x89
\x8a\x8b\x8c\x8d\x8e\x8f\x90\x91\x92\x93\x94\x95\x96\x97\x98\x99\x9a\x9b\x9c\x9d\x9e\x9f
\xa0¡¢£¤¥¦§¨©ª«¬\xad®¯°±²³´µ¶·¸¹º»¼½¾¿ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖ×ØÙÚÛÜÝÞßàáâãäåæçèéêëìíîïð
ñòóôõö÷øùúûüýþÿ'
```

Reply:
I added a warning to let the user know that this is unexpected and to check the strings.

Reply:
That's fair.